src/compiler/parser.rs

```diff
@@ -1707,20 +1762,65 @@ impl ParserType {
         self.parse_expression_or_labeled_statement()
     }
 
-    fn parse_declaration(&mut self) -> Statement {
+    fn is_declare_modifier(&self, modifier: &Node /*Modifier*/) -> bool {
+        modifier.kind() == SyntaxKind::DeclareKeyword
+    }
+
+    fn parse_declaration(&self) -> Statement {
+        let is_ambient = some(
+            self.look_ahead(|| {
+                self.parse_decorators();
+                self.parse_modifiers(None, None)
+            })
+            .as_ref()
+            .map(|node_array| {
```

Comment:
This seemed necessary to convince it to treat an `&NodeArray` as an `&[Rc<Node>]` (which it derefs to now)